### PR TITLE
Cpp code for otsu method and implementation

### PR DESCRIPTION
**Pull request by sakshiseth**
This needs changes with respect to following aspects:
- Best practice of writing code. Using in-build functions for calculations.
- Adding plots to otsu_method.cpp

### CODE REVIEW

File: otsu-method/Implementation.cpp [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	
+	// read the image in BGR format
+	Mat testImage = imread("boat.jpg", 0);
+	int bins_num = 256;
+	
+	// Get the histogram
+	long double histogram[256];
+ 
+    // initialize all intensity values to 0
+    for(int i = 0; i < 255; i++)
```
Comment (vikasgupta-github) [inline otsu-method/Implementation.cpp:19]: shouldn't this be 256? or <=255

File: otsu-method/Implementation.cpp [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+			maxi = Inter_class_variance[i];
+			getmax = i;
+		}
+	}
+	
+	cout << "Otsu's algorithm implementation thresholding result: " << bin_mids[getmax];
+
+	
+	return 0;
+}
```
Comment (vikasgupta-github) [inline otsu-method/Implementation.cpp:105]: The answer that I get from the python code is 131.982 but running the code gives 
```
Otsu's algorithm implementation thresholding result: 0.498047%  
```

File: otsu-method/Implementation.cpp [hunk 3]
```diff
@@ -0,0 +1,105 @@
+#include <iostream>
+#include "dataPath.hpp"
```
Comment (vikasgupta-github) [inline otsu-method/Implementation.cpp:2]: This is not required.

File: otsu-method/Otsu_method.cpp [hunk 4]
```diff
@@ -0,0 +1,26 @@
+// Include libraries
+#include "includeLibraries.h"
```
Comment (vikasgupta-github) [inline otsu-method/Otsu_method.cpp:2]: Not required.

File: otsu-method/dataPath.hpp [hunk 5]
```diff
@@ -0,0 +1 @@
+std::string DATA_PATH = "../data/";
```
Comment (vikasgupta-github) [inline otsu-method/dataPath.hpp:1]: This file is not required

File: otsu-method/includeLibraries.h [hunk 6]
```diff
@@ -0,0 +1,54 @@
+#pragma cling add_include_path("/usr/local/include/opencv4")
```
Comment (vikasgupta-github) [inline otsu-method/includeLibraries.h:1]: This file is not required

File: otsu-method/otsu_method.ipynb [hunk 7]
```diff
@@ -0,0 +1,119 @@
+{
```
Comment (vikasgupta-github) [inline otsu-method/otsu_method.ipynb:1]: This file is not required

**Comment by sakshiseth** [comment]
Hey @vikasgupta-github, 
I did the changes as mentioned by you here.
Let me know if this is considerable or requires any more changes.
I'll make another PR with the use of calcHist() and vector instead of simply using arrays and complex calculations.

**Pull request merged**